pkg/controller/factory/base_controller.go

```diff
@@ -77,11 +78,14 @@ func (c *baseController) Sync(ctx context.Context, syncCtx SyncContext) error {
 	return c.sync(ctx, syncCtx)
 }
 
+// QueueKey return queue key for given name.
+func QueueKey(name string) string {
+	return strings.ToLower(name) + "Key"
+}
+
 func (c *baseController) runPeriodicalResync(ctx context.Context, interval time.Duration) {
 	go wait.UntilWithContext(ctx, func(ctx context.Context) {
-		if err := c.sync(ctx, c.syncContext); err != nil {
-			utilruntime.HandleError(fmt.Errorf("periodical resync of controller %s failed: %v", c.name, err))
-		}
+		c.syncContext.Queue().Add(QueueKey(c.name))
```

Comment:
how is this distinct from the resync capabilities of informers.

Reply:
funny, I asked the same question on slack

Reply:
@sttts in many controllers we do manual time based resync... I think @deads2k asked for this at some point to make sure that when informers get stucked we still run.

Reply:
The timer used to start after the snyc finished so it was sync+1s, now its every 1s, but I don't see an issue with that.

Using the queue prevents races.

Reply:
@tnozicka if we use queue, we do time based sync, we utilize queue (so it show in metrics) and we don't race with informer. i think, this is a good change.